cw-dex-astroport/src/staking.rs

```diff
@@ -1,49 +1,66 @@
 //! Staking/rewards traits implementations for Astroport
-
+use apollo_cw_asset::AssetList;
 use apollo_utils::assets::separate_natives_and_cw20s;
-use cosmwasm_schema::cw_serde;
+use astroport::asset::{Asset as AstroAsset, Asset};
+use cosmwasm_schema::{cw_serde, QueryResponses};
 use cosmwasm_std::{
-    to_json_binary, Addr, CosmosMsg, Deps, Empty, Env, Event, QuerierWrapper, QueryRequest,
+    coins, to_json_binary, Addr, CosmosMsg, Deps, Empty, Env, Event, QuerierWrapper, QueryRequest,
     Response, Uint128, WasmMsg, WasmQuery,
 };
-use cw20::Cw20ExecuteMsg;
-
-use apollo_cw_asset::AssetList;
-use astroport::asset::Asset as AstroAsset;
-use astroport_v3::incentives::{
-    Cw20Msg as IncentivesCw20Msg, ExecuteMsg as IncentivesExecuteMsg,
-    QueryMsg as IncentivesQueryMsg,
-};
-
 use cw_dex::traits::{Rewards, Stake, Staking, Unstake};
 use cw_dex::CwDexError;
 
 /// Represents staking of tokens on Astroport
 #[cw_serde]
 pub struct AstroportStaking {
-    /// The address of the associated LP token contract
-    pub lp_token_addr: Addr,
+    /// The token factory denom of the associated LP token
+    pub lp_token_denom: String,
     /// The address of the astroport incentives contract
     pub incentives: Addr,
 }
 
+#[cw_serde]
```

Comment:
could you add a comment explaining where this comes from and why together with a permalink to the code on github